config-optional/use-external-dns-for-wildcard.yml

```diff
@@ -0,0 +1,13 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:template", "template")
+
+#@overlay/match by=overlay.subset({"kind": "Service", "metadata":{"name": "istio-ingressgateway"}})
+---
+metadata:
+  #@overlay/replace
```

Comment:
are we intentionally replacing entire annotations map with below two?

Reply:
the existing [annotation](https://github.com/cloudfoundry/cf-for-k8s/blob/master/config/_ytt_lib/github.com/cloudfoundry/cf-k8s-networking/config/istio-generated/xxx-generated-istio.yaml#L6148) is `metadata.annotations: null`  so we can't append to it.   It would need to be fixed to `metadata.annotations: {}` or similar to allow me to simply append them.   Since it tells us not to edit that generated yaml ... I have to brute force it like this.

Reply:
i see. this is what we've done somewhere else for now as an additional overlay before "real" one:

```yaml
#! Currently ytt does not allow overlaying maps on top of nils;
#! remove nil annotations before further overlays (https://github.com/k14s/ytt/issues/108)
#@overlay/match by=overlay.subset({"metadata":{"annotations": None}}),expects="1+"
---
metadata:
  #@overlay/remove
  annotations:
```

replacing all annotations on this resource may have adverse effects.

Reply:
Should we at that `remove` in somewhere so I can dump the `replace`?   or ask upstream to change it from `nil` to `{}` ?

Reply:
if above is included in cf-for-k8s repo it will be ok too.

Reply:
Added the above as an extra overlay in `config/` and removed my `replace`